service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/DeleteAzureControlledResourceStep.java

```diff
@@ -70,6 +72,18 @@ protected StepResult handleResourceDeleteException(Exception e, FlightContext co
       // retry any other non-4xx errors
       return new StepResult(AzureManagementExceptionUtils.maybeRetryStatus(ex), ex);
     }
+
+    // When retrieving shared resources from the landing zone,
+    // LZS can encounter the same management exceptions as above
+    // To detect this, we need to examine the message for those error codes
+    if (e instanceof LandingZoneServiceApiException && e.getCause() instanceof ApiException) {
+      var msg = e.getCause().getMessage();
+      if (msg != null
+          && missingResourceManagementCodes.stream().anyMatch(code -> msg.contains(code))) {
```

Comment:
Bummer that we're having to string match here. Does that mean there's potential follow-on work to make the LZS graceful in these situations? If so can you drop a ticket?

Reply:
Yeah, we could probably return a more meaningful error.